docs/advanced/runtime/docs/coop-state-machine.dot

```diff
@@ -0,0 +1,114 @@
+digraph GC {
+	label = "Coop GC State Machine";
+	labelloc = "t"
+	remincross = true;
+	
+	// Thread states
+	Starting [shape=diamond];
+	subgraph cluster_Unsafe {
+		label = "Unsafe (for GC to run)" ;
+		labelloc = "b";
+		node [color = red];
+		color = maroon;
+		style = solid;
+		Running ;
+		Self_Suspend_Requested ;
+		Async_Suspend_Requested ;
+	} ;
+	subgraph cluster_Safe {
+		labelloc = "b";
+		label = "Safe (for GC to run)" ;
+		node [color = blue];
+		color = blue;
+		style = solid;
+		Blocking ;
+		subgraph cluster_Suspended {
+			label = "" ;
+			style = invis;
+			node [shape = "rectangle", style = "filled", fillcolor = "grey95"];
+			
+			Async_Suspended ;
+			Self_Suspended ;
+			Blocking_and_Suspended [shape = "rectangle", style = "filled,solid", fillcolor="grey95"];
+		};
+	} ;
+		
+	Detached [shape=diamond] ;
+
+	// attach
+	{
+		edge [label = "attach"];
+		Starting -> Running;
+	}
+	// detach
+	{
+		edge [label = "detach"];
+		{ Running Blocking } -> Detached ;
+		// Async_Suspend_Requested -> Async_Suspend_Requested;
+	}
+	
+
+	// do blocking
+	{
+		edge [label = "do_B", color = "green", fontcolor="darkgreen"];
```

Comment:
Because I couldn't figure out how to get Graphviz not to overlap labels if they get too wide. 😢